hbase-client/src/main/java/org/apache/hadoop/hbase/client/ImmutableScan.java

```diff
@@ -0,0 +1,549 @@
+  }
+
+  @Override
+  public Scan setACL(String user, Permission perms) {
+    throw new UnsupportedOperationException("ImmutableScan does not allow access to setACL");
+  }
+
+  @Override
+  public Scan setConsistency(Consistency consistency) {
+    throw new UnsupportedOperationException(
+      "ImmutableScan does not allow access to setConsistency");
+  }
+
+  @Override
+  public Scan setReplicaId(int id) {
+    throw new UnsupportedOperationException("ImmutableScan does not allow access to setReplicaId");
+  }
+
+  @Override
+  public Scan setIsolationLevel(IsolationLevel level) {
+    throw new UnsupportedOperationException(
+      "ImmutableScan does not allow access to setIsolationLevel");
+  }
+
+  @Override
+  public Scan setPriority(int priority) {
+    throw new UnsupportedOperationException("ImmutableScan does not allow access to setPriority");
+  }
+
+  @Override
+  public Scan setScanMetricsEnabled(final boolean enabled) {
+    throw new UnsupportedOperationException(
+      "ImmutableScan does not allow access to setScanMetricsEnabled");
+  }
+
+  @Override
+  @Deprecated
+  public Scan setAsyncPrefetch(boolean asyncPrefetch) {
+    throw new UnsupportedOperationException(
+      "ImmutableScan does not allow access to setAsyncPrefetch");
+  }
+
+  @Override
+  public Scan setLimit(int limit) {
+    throw new UnsupportedOperationException("ImmutableScan does not allow access to setLimit");
+  }
+
+  @Override
+  public Scan setOneRowLimit() {
+    throw new UnsupportedOperationException(
+      "ImmutableScan does not allow access to setOneRowLimit");
+  }
+
+  @Override
+  public Scan setReadType(ReadType readType) {
+    throw new UnsupportedOperationException("ImmutableScan does not allow access to setReadType");
+  }
+
+  @Override
+  Scan setMvccReadPoint(long mvccReadPoint) {
+    throw new UnsupportedOperationException(
+      "ImmutableScan does not allow access to setMvccReadPoint");
+  }
+
+  @Override
+  Scan resetMvccReadPoint() {
+    throw new UnsupportedOperationException(
+      "ImmutableScan does not allow access to resetMvccReadPoint");
+  }
+
+  @Override
+  public Scan setNeedCursorResult(boolean needCursorResult) {
+    throw new UnsupportedOperationException(
+      "ImmutableScan does not allow access to setNeedCursorResult");
+  }
+
+  @Override
+  public long getMaxResultSize() {
+    return this.delegateScan.getMaxResultSize();
+  }
+
+  @Override
+  public Map<byte[], NavigableSet<byte[]>> getFamilyMap() {
+    return Collections.unmodifiableMap(this.delegateScan.getFamilyMap());
+  }
+
+  @Override
+  public int numFamilies() {
+    return this.delegateScan.numFamilies();
+  }
+
+  @Override
+  public boolean hasFamilies() {
+    return this.delegateScan.hasFamilies();
+  }
+
+  @Override
+  public byte[][] getFamilies() {
+    return this.delegateScan.getFamilies();
+  }
+
+  @Override
+  public byte[] getStartRow() {
```

Comment:
Just thinking whether we should be cloning this byte[]? Using this getter it is possible to really set a new start row for the scan (by direct changing byte(s) in this byte[]).

Reply:
Yeah, makes sense. Done